en/guide/missions.md

```diff
@@ -0,0 +1,201 @@
+
+The code fragment below shows how this is done, using promises to wait on the result.
+
+```cpp
+{
+    auto prom = std::make_shared<std::promise<Mission::Result>>();
+    auto future_result = prom->get_future();
+    device.mission().start_mission_async(
+    [prom](Mission::Result result) {
+        prom->set_value(result);
+    });
+
+    const Mission::Result result = future_result.get(); //Wait on result
+    if (result != Mission::Result::SUCCESS) {
+        std::cout << "Mission start failed (" << Mission::result_str(result) << "), exiting." << std::endl;
+        return 1;
+    }
+    std::cout << "Started mission." << std::endl;
+}
+```
+
+To pause a mission use [Mission::pause_mission_async()](../api_reference/classdronecore_1_1_mission.md#classdronecore_1_1_mission_1a65f729cf954586507ecd8dc07a510dd1). The code is almost exactly the same as for starting a mission:
+
+```cpp
+{
+    auto prom = std::make_shared<std::promise<Mission::Result>>();
+    auto future_result = prom->get_future();
+
+    std::cout << "Pausing mission..." << std::endl;
+    device.mission().pause_mission_async(
+    [prom](Mission::Result result) {
+        prom->set_value(result);
+    });
+
+    const Mission::Result result = future_result.get();
+    if (result != Mission::Result::SUCCESS) {
+        std::cout << "Failed to pause mission (" << Mission::result_str(result) << ")" << std::endl;
+    } else {
+        std::cout << "Mission paused." << std::endl;
+    }
+}
+```
+
+## Monitoring Progress
+
+Asyncronously monitor progress using [Mission::subscribe_progress()](../api_reference/classdronecore_1_1_mission.md#classdronecore_1_1_mission_1a3290fc79eb22f899528328adfca48a61), which receives a regular callback with the current `MissionItem` number and the total number of items.
+
+The code fragment just takes a lambda function that reports the current status. Note that the mission is complete when `current == total`.
+
+```cpp
+device.mission().subscribe_progress( [](int current, int total) {
+       std::cout << "Mission status update: " << current << " / " << total << std::endl;
+    });
+```
+
+The following synchronous methods are also available for checking mission progress:
+* [mission_finished()](../api_reference/classdronecore_1_1_mission.md#classdronecore_1_1_mission_1abf3463efaa18147a1c179e7449503829) - Checks if mission has been finished.
+* [current_mission_item()](../api_reference/classdronecore_1_1_mission.md#classdronecore_1_1_mission_1a1faa448b32cd0028923b22de0cc78e9c) - Returns the current mission item index.
+* [total_mission_items()](../api_reference/classdronecore_1_1_mission.md#classdronecore_1_1_mission_1a9d2195ec1af301c51002f8cb99aa22e9) - Gets the total number of items.

```

Comment:
You can add hat if `current_mission_item() == total_mission_items()` this also means that the mission is finished.